include/toolchain/common.h

```diff
@@ -106,6 +106,12 @@
 #define _DO_CONCAT(x, y) x ## y
 #define _CONCAT(x, y) _DO_CONCAT(x, y)
 
+/* Additionally used as a sentinel by gen_syscalls.py to identify what
+ * functions are system calls
+ */
+#define __syscall static inline
+#define __syscall_inline static inline
```

Comment:
If `__syscall` and `__syscall_inline` both expand to the same thing, why have both of them? Are there plans to make `__syscall` expand to something else?

Reply:
they have different semantics in gen_syscalls.py. __syscall_inline causes the _INLINE variant of K_SYSCALL_DECLARE to be used